examples/pytorch/mixhop/main.py

```diff
@@ -0,0 +1,322 @@
+                    output = self.weights[str(j)](feats)
+                    outputs.append(output)
+
+                feats = feats * norm
+                graph.ndata['h'] = feats
+                graph.update_all(fn.copy_u('h', 'm'), fn.sum('m', 'h'))
+                feats = graph.ndata.pop('h')
+                feats = feats * norm
+            
+            final = torch.cat(outputs, dim=1)
+            
+            if self.batchnorm:
+                final = self.bn(final)
+            
+            if self.activation is not None:
+                final = self.activation(final)
+            
+            final = self.dropout(final)
+
+            return final
+
+class MixHop(nn.Module):
+    def __init__(self,
+                 in_dim,
+                 hid_dim, 
+                 out_dim,
+                 num_layers=2,
+                 p=[0, 1, 2],
+                 input_dropout=0.0,
+                 layer_dropout=0.0,
+                 activation=None,
+                 batchnorm=False):
+        super(MixHop, self).__init__()
+        self.in_dim = in_dim
+        self.hid_dim = hid_dim
+        self.out_dim = out_dim
+        self.num_layers = num_layers
+        self.p = p
+        self.input_dropout = input_dropout
+        self.layer_dropout = layer_dropout
+        self.activation = activation
+        self.batchnorm = batchnorm
+
+        self.layers = nn.ModuleList()
+        self.dropout = nn.Dropout(self.input_dropout)
+
+        # Input layer
+        self.layers.append(MixHopConv(self.in_dim,
+                                      self.hid_dim,
+                                      p=self.p,
+                                      dropout=self.input_dropout,
+                                      activation=self.activation,
+                                      batchnorm=self.batchnorm))
+        
+        # Hidden layers with n - 1 MixHopConv layers
+        for i in range(self.num_layers - 2):
+            self.layers.append(MixHopConv(self.hid_dim * len(args.p),
+                                          self.hid_dim,
+                                          p=self.p,
+                                          dropout=self.layer_dropout,
+                                          activation=self.activation,
+                                          batchnorm=self.batchnorm))
+        
+        self.fc_layers = nn.Linear(self.hid_dim * len(args.p), self.out_dim, bias=False)
+
+    def forward(self, graph, feats):
+        feats = self.dropout(feats)
+        for layer in self.layers:
+            feats = layer(graph, feats)
+        
+        feats = self.fc_layers(feats)
+
+        return feats
+
+def setup_seed(seed):
+    torch.manual_seed(seed)
+    torch.cuda.manual_seed_all(seed)
+    np.random.seed(seed)
+    random.seed(seed)
+    torch.backends.cudnn.deterministic = True
+
+def main(args):
+    # Step 0: set random seed
+    if args.seed >= 0:
+        setup_seed(args.seed)
```

Comment:
Let's remove the seeding part. Otherwise, 100 runs make no sense.

Reply:
Done.